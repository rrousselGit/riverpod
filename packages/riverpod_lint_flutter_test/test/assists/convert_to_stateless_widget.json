[{"priority":34,"change":{"message":"Convert to StatelessWidget","edits":[{"fileStamp":0,"edits":[{"offset":520,"length":10,"replacement":"StatelessWidget"}]}],"linkedEditGroups":[]}},{"priority":34,"change":{"message":"Convert to StatelessWidget","edits":[{"fileStamp":0,"edits":[{"offset":787,"length":23,"replacement":""},{"offset":678,"length":18,"replacement":"StatelessWidget"}]}],"linkedEditGroups":[]}},{"priority":34,"change":{"message":"Convert to StatelessWidget","edits":[{"fileStamp":0,"edits":[{"offset":1676,"length":142,"replacement":""},{"offset":1673,"length":0,"replacement":"@override\n  Widget build(BuildContext context) {\n    return const Placeholder();\n  }\n"},{"offset":1603,"length":69,"replacement":""},{"offset":1544,"length":18,"replacement":"StatelessWidget"}]}],"linkedEditGroups":[]}},{"priority":34,"change":{"message":"Convert to StatelessWidget","edits":[{"fileStamp":0,"edits":[{"offset":2008,"length":159,"replacement":""},{"offset":2005,"length":0,"replacement":"@override\n  Widget build(BuildContext context) {\n    return const Placeholder();\n  }\n"},{"offset":1918,"length":86,"replacement":""},{"offset":1851,"length":22,"replacement":"StatelessWidget"}]}],"linkedEditGroups":[]}},{"priority":34,"change":{"message":"Convert to StatelessWidget","edits":[{"fileStamp":0,"edits":[{"offset":2383,"length":167,"replacement":""},{"offset":2380,"length":0,"replacement":"@override\n  Widget build(BuildContext context) {\n    return const Placeholder();\n  }\n"},{"offset":2279,"length":100,"replacement":""},{"offset":2204,"length":26,"replacement":"StatelessWidget"}]}],"linkedEditGroups":[]}},{"priority":34,"change":{"message":"Convert to StatelessWidget","edits":[{"fileStamp":0,"edits":[{"offset":2676,"length":23,"replacement":""},{"offset":2575,"length":14,"replacement":"StatelessWidget"}]}],"linkedEditGroups":[]}}]