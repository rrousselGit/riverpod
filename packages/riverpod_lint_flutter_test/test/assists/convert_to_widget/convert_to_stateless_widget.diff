Message: `Convert to StatelessWidget`
Priority: 34
Diff for file `test/assists/convert_to_widget/convert_to_widget.dart:25`:
```
}

- class Hook extends HookWidget {
+ class Hook extends StatelessWidget {
  const Hook({super.key});

```
---
Message: `Convert to StatelessWidget`
Priority: 34
Diff for file `test/assists/convert_to_widget/convert_to_widget.dart:34`:
```
}

- class HookConsumer extends HookConsumerWidget {
-   const HookConsumer({super.key});
- 
-   @override
-   Widget build(
-     BuildContext context,
-     WidgetRef ref,
-   ) {
+ class HookConsumer extends StatelessWidget {
+   const HookConsumer({super.key});
+ 
+   @override
+   Widget build(
+     BuildContext context) {
    return const Placeholder();
  }
```
---
Message: `Convert to StatelessWidget`
Priority: 34
Diff for file `test/assists/convert_to_widget/convert_to_widget.dart:78`:
```
}

- class HookStateful extends StatefulHookWidget {
-   const HookStateful({super.key});
- 
-   @override
-   State<HookStateful> createState() => HookStatefulState();
- }
- 
- class HookStatefulState extends State<HookStateful> {
+ class HookStateful extends StatelessWidget {
+   const HookStateful({super.key});
+ 
+  
  @override
  Widget build(BuildContext context) {
```
---
Message: `Convert to StatelessWidget`
Priority: 34
Diff for file `test/assists/convert_to_widget/convert_to_widget.dart:92`:
```
}

- class ConsumerStateful extends ConsumerStatefulWidget {
-   const ConsumerStateful({super.key});
- 
-   @override
-   ConsumerState<ConsumerStateful> createState() => _ConsumerStatefulState();
- }
- 
- class _ConsumerStatefulState extends ConsumerState<ConsumerStateful> {
+ class ConsumerStateful extends StatelessWidget {
+   const ConsumerStateful({super.key});
+ 
+  
  @override
  Widget build(BuildContext context) {
```
---
Message: `Convert to StatelessWidget`
Priority: 34
Diff for file `test/assists/convert_to_widget/convert_to_widget.dart:106`:
```
}

- class HookConsumerStateful extends StatefulHookConsumerWidget {
-   const HookConsumerStateful({super.key});
- 
-   @override
-   ConsumerState<HookConsumerStateful> createState() =>
-       _HookConsumerStatefulState();
- }
- 
- class _HookConsumerStatefulState extends ConsumerState<HookConsumerStateful> {
+ class HookConsumerStateful extends StatelessWidget {
+   const HookConsumerStateful({super.key});
+ 
+  
  @override
  Widget build(BuildContext context) {
```
---
Message: `Convert to StatelessWidget`
Priority: 34
Diff for file `test/assists/convert_to_widget/convert_to_widget.dart:121`:
```
}

- class Consumer extends ConsumerWidget {
-   const Consumer({super.key});
- 
-   @override
-   Widget build(
-     BuildContext context,
-     WidgetRef ref,
-   ) {
+ class Consumer extends StatelessWidget {
+   const Consumer({super.key});
+ 
+   @override
+   Widget build(
+     BuildContext context) {
    return const Placeholder();
  }
```
---
Message: `Convert to StatelessWidget`
Priority: 34
Diff for file `test/assists/convert_to_widget/convert_to_widget.dart:153`:
```
}

- class HookConsumerWithField extends HookConsumerWidget {
-   const HookConsumerWithField({
-     super.key,
-     required this.field,
-   });
- 
-   final int field;
-   static final int staticField = 42;
- 
-   @override
-   Widget build(
-     BuildContext context,
-     WidgetRef ref,
-   ) {
+ class HookConsumerWithField extends StatelessWidget {
+   const HookConsumerWithField({
+     super.key,
+     required this.field,
+   });
+ 
+   final int field;
+   static final int staticField = 42;
+ 
+   @override
+   Widget build(
+     BuildContext context) {
    return Column(
      children: [
```
---
Message: `Convert to StatelessWidget`
Priority: 34
Diff for file `test/assists/convert_to_widget/convert_to_widget.dart:180`:
```
}

- class ConsumerStatefulWithField extends ConsumerStatefulWidget {
-   const ConsumerStatefulWithField({
-     super.key,
-     required this.field,
-     required this.foo,
-   });
- 
-   final int field;
-   final FooClass foo;
-   static final int staticField = _constantNumber;
- 
-   @override
-   ConsumerState<ConsumerStatefulWithField> createState() =>
-       _ConsumerStatefulWithFieldState();
- }
- 
- const _constantNumber = 42;
- 
- class _ConsumerStatefulWithFieldState
-     extends ConsumerState<ConsumerStatefulWithField> {
-   void printFoo() {
-     print(widget.foo);
-   }
- 
-   @override
-   Widget build(BuildContext context) {
-     print(widget.field);
-     printFoo();
-     return Column(
-       children: [
-         Text('${widget.field}'),
-         Text('${widget.foo.bar}'),
-         Text('${ConsumerStatefulWithField.staticField}'),
-       ],
-     );
-   }
- }
+ class ConsumerStatefulWithField extends StatelessWidget {
+   const ConsumerStatefulWithField({
+     super.key,
+     required this.field,
+     required this.foo,
+   });
+ 
+   final int field;
+   final FooClass foo;
+   static final int staticField = _constantNumber;
+ 
+  
+   void printFoo() {
+     print(foo);
+   }
+ 
+   @override
+   Widget build(BuildContext context) {
+     print(field);
+     printFoo();
+     return Column(
+       children: [
+         Text('$field'),
+         Text('${foo.bar}'),
+         Text('$staticField'),
+       ],
+     );
+   }
+ }
+ 
+ const _constantNumber = 42;

```
---
