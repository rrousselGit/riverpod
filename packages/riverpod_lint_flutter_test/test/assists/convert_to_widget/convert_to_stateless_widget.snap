Message: `Convert to StatelessWidget`
<<<< start: 520 -- end: 523
Hoo
==== start: 520 -- end: 528
Stateles
>>>>
Message: `Convert to StatelessWidget`
<<<< start: 678 -- end: 809
HookConsumerWidget {
  const HookConsumer({super.key});

  @override
  Widget build(
    BuildContext context,
    WidgetRef ref,
 
==== start: 678 -- end: 783
StatelessWidget {
  const HookConsumer({super.key});

  @override
  Widget build(
    BuildContext contex
>>>>
Message: `Convert to StatelessWidget`
<<<< start: 1549 -- end: 1817
fulHookWidget {
  const HookStateful({super.key});

  @override
  State<HookStateful> createState() => HookStatefulState();
}

class HookStatefulState extends State<HookStateful> {
  @override
  Widget build(BuildContext context) {
    return const Placeholder();
  }

==== start: 1549 -- end: 1688
lessWidget {
  const HookStateful({super.key});

  
@override
  Widget build(BuildContext context) {
    return const Placeholder();
  }
}

>>>>
Message: `Convert to StatelessWidget`
<<<< start: 1851 -- end: 2166
ConsumerStatefulWidget {
  const ConsumerStateful({super.key});

  @override
  ConsumerState<ConsumerStateful> createState() => _ConsumerStatefulState();
}

class _ConsumerStatefulState extends ConsumerState<ConsumerStateful> {
  @override
  Widget build(BuildContext context) {
    return const Placeholder();
  }

==== start: 1851 -- end: 1999
StatelessWidget {
  const ConsumerStateful({super.key});

  
@override
  Widget build(BuildContext context) {
    return const Placeholder();
  }
}

>>>>
Message: `Convert to StatelessWidget`
<<<< start: 2209 -- end: 2549
fulHookConsumerWidget {
  const HookConsumerStateful({super.key});

  @override
  ConsumerState<HookConsumerStateful> createState() =>
      _HookConsumerStatefulState();
}

class _HookConsumerStatefulState extends ConsumerState<HookConsumerStateful> {
  @override
  Widget build(BuildContext context) {
    return const Placeholder();
  }

==== start: 2209 -- end: 2356
lessWidget {
  const HookConsumerStateful({super.key});

  
@override
  Widget build(BuildContext context) {
    return const Placeholder();
  }
}

>>>>
Message: `Convert to StatelessWidget`
<<<< start: 2575 -- end: 2698
ConsumerWidget {
  const Consumer({super.key});

  @override
  Widget build(
    BuildContext context,
    WidgetRef ref,
 
==== start: 2575 -- end: 2676
StatelessWidget {
  const Consumer({super.key});

  @override
  Widget build(
    BuildContext contex
>>>>
